os/drivers/ai-soc/ndp120/src/ndp120_api.c

```diff
@@ -77,6 +77,10 @@
 /* currently the max size of tank is limited to ~580ms  */
 #define AUDIO_BEFORE_MATCH_MS	(1500)
 
+/* when defined, the let the NDP use an external PDM clock */
+#define USE_EXTERNAL_PDM_CLOCK
```

Comment:
Is there any case we do not use external PDM clock? we should use for all the time, am I right?
Even if there are some case we should not use external PDM clock, actually we should check gpio pin you defined in above file instead of constant value.
```
#ndo120.h
struct ndp120_lower_s {
	/* SPI related configs */
        struct spi_io_config spi_config;

	/* callback function: attach the ndp120 interrupt handler to the GPIO interrupt */
        CODE int (*attach)(ndp120_handler_t handler, FAR char *arg);

	/* eanble/disable ndp120 interrupts hadling on rlt8730e */
	CODE void (*irq_enable)(bool enable);
	CODE void (*external_pdm_enabled)(void); // new function
};

#define EXTERNAL_PDM_CLOCK_RATE 1536000
#define INTERNAL_PDM_CLOCK_RATE 768000
static int configure_audio(struct syntiant_ndp_device_s *ndp, unsigned int pdm_in_shift,
				unsigned int audio_tank_ms, unsigned int *audio_frame_bytes)
{
	memset(&pdm_config, 0, sizeof(pdm_config));
	
	/* if external pdm need to be used, then change clk mode * pdm rate */
	if (ndp->dev->lower->external_pdm_enabled()) {
		pdm_config.clk_mode = SYNTIANT_NDP120_CONFIG_VALUE_PDM_CLK_MODE_EXTERNAL;
		pdm_config.pdm_rate = EXTERNAL_PDM_CLOCK_RATE;
	} else {
		pdm_config.clk_mode = SYNTIANT_NDP120_CONFIG_VALUE_PDM_CLK_MODE_DUAL_INTERNAL;
		pdm_config.pdm_rate = INTERNAL_PDM_CLOCK_RATE;
	}
	pdm_config.interface = 0;
	pdm_config.clk = SYNTIANT_NDP120_CONFIG_VALUE_PDM_CLK_ON;
	pdm_config.sample_rate = 16000;
	....
}
```
I think this is proper approach.

Reply:
From a couple of discussions (also with @abhishek-samsung) it sounds like the optimal approach is to integrate the selection of external (I2S BCLK) vs internal (NDP generated) PDM clock with the PM module, where we can register callbacks to be called when entering sleep state and normal/wakeup state respectively. 

My suggestion would be to keep the current behaviour (where it's controlled by a #define) in this PR, and then we can add a new PR with the relevant changes with PM callbacks subsequently.
Is the PM code for RTL8730e in a functional state where we already implement and test this?

Reply:
Hello @cvejlbo , previously we got a request regarding the integration of I2S BCLCK and NDP PDM CLOCK.
This PR is doing the necessary operation as per our understanding, please check https://github.com/Samsung/TizenRT/pull/6304.
But, it was yet to verified with the NDP, could you help to review the PR and check whether it meets the requirement?

Reply:
Hello @cvejlbo , yes. PM related changes are taking a bit more time than expected. Also, we havent enabled pm on product code yet for bixby. So, the current PR would be okay to merge.

@edwakuwaku , in PR #6304 , we are only setting the gpio pin state. However, we also need to run some api's from NDP side in addition to changing the gpio pin state. As a result, that PR is not fully enough. Currently PM related engineers are working on verifying the callback feature in PM. Once its done, I think, we should combine the changes in PR #6304 + ndp changes and add them to callbacks.

Reply:
Hello @abhishek-samsung , thanks for updating~